client/my-sites/current-site/index.jsx

```diff
@@ -159,9 +156,13 @@ const CurrentSite = React.createClass( {
 // TODO: make this pure when sites can be retrieved from the Redux state
 module.exports = connect(
 	( state, ownProps ) => {
+		const selectedSiteId = getSelectedSiteId( state );
 		const selectedSite = getSelectedSite( state );
+
 		return {
-			selectedSite: selectedSite || ownProps.sites.getPrimary()
+			selectedSiteId,
```

Comment:
> When would it ever be the case that we'd want to defer to the primary site and not the "selected" site?

Hmm, I can't think of any routes that do that.

In that case, I wonder if we introduced a regression here. Navigating to http://wordpress.com/pages for example will render the primary site rather than all sites